File: frontend/src/types/expiry.ts
export interface InventoryBatch {
  id: number
  product_id: number
  batch_number: string
  expiry_date: string
  quantity: number
  received_date: string
  is_disposed: boolean
  days_to_expiry: number
  expiry_status: 'valid' | 'expiring_soon' | 'expired'
  product: { id: number; name: string; sku: string; category: { id: number; name: string } }
}

export interface ExpirySummary {
  expiring_soon: number
  expired: number
  valid: number
  expiring_within_days: number
}

export interface Notification {
  id: number
  type: string
  title: string
  message: string
  reference?: string
  is_read: boolean
  created_at: string
}

export interface Paged<T> {
  items: T[]
  total: number
  page: number
  page_size: number
}

File: backend/app/schemas/expiry.py
from datetime import date, datetime

from pydantic import BaseModel, ConfigDict, Field

from app.schemas.catalog import ProductResponse


class BatchCreate(BaseModel):
    product_id: int
    batch_number: str = Field(min_length=1, max_length=80)
    expiry_date: date
    quantity: int = Field(gt=0)


class BatchResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    product_id: int
    batch_number: str
    expiry_date: date
    quantity: int
    received_date: date
    is_disposed: bool
    days_to_expiry: int
    expiry_status: str
    product: ProductResponse


class ExpirySummary(BaseModel):
    expiring_soon: int
    expired: int
    valid: int
    expiring_within_days: int


class NotificationResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    type: str
    title: str
    message: str
    reference: str | None = None
    is_read: bool
    created_at: datetime


class DisposeBatchRequest(BaseModel):
    reason: str = Field(default="Expired stock disposed", min_length=2)
